src/network_test.go

```diff
@@ -0,0 +1,101 @@
+package cache
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"./cache"
+	"./datastore"
+)
+
+type NetTestConfig struct {
+	cn CacheNet
+}
+
+var CONFIG NetTestConfig = NetTestConfig{
+	cn: CacheNet{},
+}
+
+func TestBasicNetworkCall(t *testing.T) {
+	fmt.Printf("TestBasicNetworkCall ...\n")
+	failed := false
+	data := datastore.MakeDataStore()
+
+	// add files to datastore
+	for j := 0; j < 5; j++ {
+		filename := "fake_" + strconv.Itoa(j) + ".txt"
+		data.Make(filename)
+	}
+
+	// this copies data, so can't adjust later
+	var lruCache cache.Cache
+	lruCache.Init(1, cache.CACHE_SIZE, cache.LRU, data)
+	sockname := CONFIG.cn.startCacheRPCServer(&lruCache)
+
+	for j := 0; j < 5; j++ {
+		filename := "fake_" + strconv.Itoa(j) + ".txt"
+		args := cache.RequestFileArgs{Filename: filename}
+		reply := cache.RequestFileReply{}
+
+		ok := call(sockname, "Cache.FetchRPC", &args, &reply)
+		if !ok {
+			t.Errorf("Could not open %s from cache", filename)
+			failed = true
+		}
+	}
+
+	if failed {
+		fmt.Printf("\t... FAILED\n")
+	} else {
+		fmt.Printf("\t... PASSED\n")
+	}
+}
+
+func TestDoubleNetworkCall(t *testing.T) {
+	fmt.Printf("TestDoubleNetworkCall ...\n")
+	failed := false
+	data := datastore.MakeDataStore()
+
+	// add files to datastore
+	for j := 0; j < 5; j++ {
+		filename := "fake_" + strconv.Itoa(j) + ".txt"
+		data.Make(filename)
+	}
+
+	// this copies data, so can't adjust later
+	var lruCacheFirst cache.Cache
+	lruCacheFirst.Init(1, cache.CACHE_SIZE, cache.LRU, data)
+	socknameFirst := CONFIG.cn.startCacheRPCServer(&lruCacheFirst)
+
+	var lruCacheSecond cache.Cache
+	lruCacheSecond.Init(1, cache.CACHE_SIZE, cache.LRU, data)
```

Comment:
How does this pass? The IDs are the same for both?

Reply:
ohhh IDs are not actually being checked anywhere / nothing is being done with them except at the cache master level, which there are no tests for right now :/

Reply:
edited it for now, but we will need to code to something to check these things at some point.